src/main/java/io/xpring/ilp/grpc/IlpCredentials.java

```diff
@@ -0,0 +1,72 @@
+package io.xpring.ilp.grpc;
+
+import com.google.common.base.Preconditions;
+import io.grpc.CallCredentials;
+import io.grpc.Metadata;
+import io.grpc.Status;
+import io.xpring.xrpl.XpringException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Executor;
+
+/**
+ * An extension of {@link CallCredentials} which provides a convenient way to
+ * add an Authorization metadata header, and ensures every bearer token
+ * going over the wire is prefixed with 'Bearer '
+ */
+public class IlpCredentials extends CallCredentials {
+
+  public static final String BEARER_SPACE = "Bearer ";
```

Comment:
nit: `BEARER_WITH_SPACE` is more verbose but slightly clearer. Otherwise it sounds like "bearer space" is a noun. You could also use `BEARER_PREFIX`